apps/base-docs/tutorials/docs/0_gassless-transactions-with-paymaster.md

```diff
@@ -0,0 +1,616 @@
+
+### Test your Paymaster policy
+
+We will now test the policy that was created.
+
+:::info The Use of Foundry
+
+In this tutorial, we use foundry to create two key pairs that will allow us to create Smart Accounts that we will sponsor transactions on behalf of. You will **not** need to send any funds (ETH) to these wallets. You may use private keys of wallets you own as an alternative.
+
+Foundry depends on Rust to work. If you do not have rust install by running `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
+
+For more information, see the [Foundry Book installation guide](https://book.getfoundry.sh/getting-started/installation).
+
+:::
+
+**Open a terminal and install Foundry by running:**
+
+```bash
+curl -L https://foundry.paradigm.xyz | bash
+```
+
+**Install the Foundry toolchain installer:**
+
+```bash
+foundryup
+```
+
+**Create a directory named `sponosored_transactions`:**
+
+```bash
+mkdir sponosored_transactions
+```
+
+**Change into the new directory, initialize a node project, install two dependencies `viem` and `permissionless` and create a file titled `index.js`:**
+
+```bash
+cd paymaster-tutorial
+npm init es6
+npm install permissionless
+npm install viem
+touch index.js
+```
+
+In the `index.js` file, import the dependencies from viem and permissionless.
+These dependencies will allow us to connect to Base, create our smart accounts, initialize a paymaster, and send our encoded data to the network.
+
+```javascript
+import { http, createPublicClient, encodeFunctionData } from 'viem';
+import { base } from 'viem/chains';
+import { createSmartAccountClient } from 'permissionless';
+import { privateKeyToSimpleSmartAccount } from 'permissionless/accounts';
+import { createPimlicoPaymasterClient } from 'permissionless/clients/pimlico';
+```
+
+### Setting up the constants for entrypoint and rpc url
+
+:::note Find your RPC URL
+
+The RPC URL is the URL for your Coinbase Developer Platform (CDP) Paymaster.
+
+This was saved in the previous section and follows this format: `https://api.developer.coinbase.com/rpc/v1/base/<SPECIAL-KEY>`
+
+Navigate to the [Paymaster Tool](https://portal.cdp.coinbase.com/products/bundler-and-paymaster) and select the `Configuration` tab at the top of the screen to obtain your RPC URL.
+
+:::
+
+We will create a constant for our RPC url obtained from cdp.portal.coinbase.com. The most secure way to do this is by using a proxy. For the purposes of this demo we will hardcode it into our `index.js` file. For product, we highly recommend using a [proxy service](https://www.smartwallet.dev/guides/paymasters).
+
+We will also need the address of the entrypoint contract for Base. A full list of entrpoint contracts and their addresses can be found [here](https://docs.alchemy.com/reference/factory-addresses).
+
+**Add the following to lines of code after your import statements:**
+
+```javascript
```

Comment:
Should these three be a single code block?

Reply:
Yeah makes it easy for them to copy and paste